src/bios.rs

```diff
@@ -157,10 +159,66 @@ impl Component for Bios {
         crate::component::query_adopt_state()
     }
 
+    // Backup the current grub.cfg and install new static config
+    // - Backup "/boot/loader/grub.cfg" to "/boot/grub2/grub.cfg.bak"
+    // - Remove symlink "/boot/grub2/grub.cfg"
+    // - Install new static grub.cfg
+    // - Remove old "/boot/loader/grub.cfg"
+    fn migrate_static_grub_config(&self, sysroot_path: &str, destdir: &openat::Dir) -> Result<()> {
+        use anyhow::Context;
+        use openat_ext::OpenatDirExt;
+
+        // sysroot_path is /, destdir is Dir of /
+        let grub_config_path = Utf8PathBuf::from(sysroot_path).join("boot/grub2");
+        let grub_config_dir = destdir
+            .sub_dir("boot/grub2")
+            .context("Opening boot/grub2")?;
+
+        let grub_config = grub_config_path.join(grubconfigs::GRUBCONFIG);
+
+        // On BIOS, /boot/grub2/grub.cfg is symlink to /boot/loader/grub.cfg,
+        // backup it to /boot/grub2/grub.cfg.bak
+        if !grub_config.exists() {
+            anyhow::bail!("Not found '{}'", grub_config);
+        } else if !grub_config.is_symlink() {
+            anyhow::bail!("'{}' is not a symlink", grub_config);
+        } else {
```

Comment:
Minor/optional: mentioned this in a previous comment; since both paths above return, we don't strictly need this in an else block. That allows us to outdent one level.

Reply:
Got it, sorry for the misunderstooding.